src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -163,6 +163,19 @@ void GcsResourceManager::UpdateFromResourceReport(const rpc::ResourcesData &data
   UpdateNodeResourceUsage(node_id, data);
 }
 
+void GcsResourceManager::UpdateResourceLoads(const rpc::ResourcesData &data) {
+  NodeID node_id = NodeID::FromBinary(data.node_id());
+  auto iter = node_resource_usages_.find(node_id);
+  if (iter == node_resource_usages_.end()) {
+    // It will happen when the node has been deleted or hasn't been added.
+    return;
+  }
+  if (data.resource_load_changed()) {
+    (*iter->second.mutable_resource_load()) = data.resource_load();
+  }
+  (*iter->second.mutable_resource_load_by_shape()) = data.resource_load_by_shape();
```

Comment:
this line should be moved into the if statement. (I know you just copy them, but let's fix it by the way)

Reply:
Hmmm, I'm not sure. But I see no damage there. The lightweight resource broadcasting is not fault-tolerant I think and if we missed some rpc request, it's going to be broken for some cases for ever.

Reply:
It will be inconsistent with `resource_load`